cmd/main.go

```diff
@@ -75,77 +99,76 @@ func init() {
 
 	rootCmd.DisableAutoGenTag = true
 	rootCmd.SetOut(os.Stdout)
+	return rootCmd, nil
 }
 
-func main() {
-	err := rootCmd.Execute()
-
-	if err == nil {
-		os.Exit(0)
-	}
-
-	logger.Failuref("%v", err)
-	os.Exit(1)
-}
-
-func run(cmd *cobra.Command, args []string) error {
+func run(kubeconfigArgs *genericclioptions.ConfigFlags, logger stderrLogger, flags args, printFlags *k8sget.PrintFlags) error {
 	if flags.version {
 		fmt.Printf(`{"version":"%s","sha":"%s","date":"%s"}`+"\n", version, commit, date)
 		return nil
 	}
 
-	logger = stderrLogger{
-		stderr:  os.Stderr,
-		verbose: flags.verbose,
+	if !flags.verbose {
+		klog.LogToStderr(false)
 	}
 
-	cfg, err := kubeconfigArgs.ToRESTConfig()
+	// default processing
+	gitopsDynClient, err := getDynClient(kubeconfigArgs)
 	if err != nil {
 		return err
 	}
 
-	cfg.WarningHandler = rest.NoWarnings{}
-
-	disc, err := discovery.NewDiscoveryClientForConfig(cfg)
+	clusterDiscoveryClient, err := getDiscoveryClient(kubeconfigArgs)
 	if err != nil {
 		return err
 	}
 
-	_, list, err := disc.ServerGroupsAndResources()
+	clusterDynClient, err := getDynClient(kubeconfigArgs)
 	if err != nil {
 		return err
 	}
 
-	dynClient, err := dynamic.NewForConfig(cfg)
+	clusterRestClient, err := getRestClient(kubeconfigArgs)
 	if err != nil {
 		return err
 	}
 
-	cfg.GroupVersion = &ksapi.GroupVersion
-	var scheme = runtime.NewScheme()
-	ksapi.AddToScheme(scheme)
-	var codecs = serializer.NewCodecFactory(scheme)
-	cfg.NegotiatedSerializer = codecs.WithoutConversion()
-	cfg.APIPath = "/apis"
-
-	structClient, err := rest.RESTClientFor(cfg)
+	zombies, err := detectZombies(logger, flags, gitopsDynClient, clusterDynClient, clusterDiscoveryClient, clusterRestClient, *kubeconfigArgs.Namespace)
 	if err != nil {
 		return err
 	}
 
-	helmReleases, err := listResources(context.TODO(), dynClient.Resource(schema.GroupVersionResource{
-		Group:    "helm.toolkit.fluxcd.io",
-		Version:  "v2beta1",
-		Resource: "helmreleases",
-	}))
+	return printZombies(zombies, printFlags)
+}
+
+func detectZombies(logger stderrLogger, flags args, gitopsDynClient, clusterDynClient dynamic.Interface, clusterDiscoveryClient *discovery.DiscoveryClient, clusterRestClient *rest.RESTClient, namespace string) ([]unstructured.Unstructured, error) {
+	var zombies []unstructured.Unstructured
 
+	logger.Debugf("⎈ Helm releases ⎈")
+	helmReleases, err := listHelmReleases(context.TODO(), gitopsDynClient, flags)
 	if err != nil {
-		return fmt.Errorf("failed to get helmreleases: %w", err)
+		return nil, fmt.Errorf("failed to get helmreleases: %w", err)
+	}
+	for _, h := range helmReleases {
+		logger.Debugf(h.GetName())
 	}
 
-	kustomizations, err := listKustomizations(context.TODO(), structClient)
+	logger.Debugf("👷 Kustomizations 👷")
```

Comment:
Same here, `discover all kustomizations`.

Also I don't have anything against emojis, but I reckon its more professional without them, especially in error types (like used in some other parts of the pr).